src/components/charts/PercentageStackedBarChart.jsx

```diff
@@ -0,0 +1,114 @@
+import {
+  BarChart, Bar, Tooltip, XAxis, YAxis,
+  CartesianGrid, LabelList,
+} from 'recharts';
+import PropTypes from 'prop-types';
+import React from 'react';
+import _ from 'underscore';
+import ChartsHelper from './ChartsHelper';
+import './PercentageStackedBarChart.less';
+
+const getPercentageDataLabels = chartData => chartData.map(entry => entry.name);
+
+const xAxisStyle = {
+  fontSize: '10px',
+  fontWeight: 600,
+  lineHeight: '1em',
+  letterSpacing: '.02rem',
+  color: '#3283c8',
+};
+
+const labelListStyle = {
+  fill: '#FFFFFF',
+  fontSize: '14px',
+  fontWeight: 600,
+};
+
+class PercentageStackedBarChart extends React.Component {
+  render() {
+    const percentageData = ChartsHelper.getPercentageData(
+      this.props.data,
+      this.props.percentageFixedPoint,
+    );
+    const percentageDataLabels = getPercentageDataLabels(this.props.data);
+    return (
+      <div className="percentage-bar-chart">
+        <BarChart
+          width={580}
```

Comment:
will we ever want to change these values?

Reply:
I agree with you!!! To me at first glance this design looks relatively special for this case (e.g. only one bar with all stacked items, 2 columns of labels in legend) so I haven't think much about which parameters shall we make it customize. But as you said we can first make the left width, height, and column number customizable~  😛